RepoDb/RepoDb/Extensions/ExpressionExtension.cs

```diff
@@ -74,9 +74,80 @@ internal static bool IsMathematical(this Expression expression)
-            var type = expression.GetMemberType();
-            return new Field(name,type);
+            if (expression.Left.IsMember())
+            {
+                return expression.Left.ToMember().GetField();
+            }
+            else if (expression.Left.IsUnary())
+            {
+                return expression.Left.ToUnary().GetField();
+            }
+            throw new NotSupportedException($"Expression '{expression.ToString()}' is currently not supported.");
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Field"/> defined on the current instance of <see cref="UnaryExpression"/>
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+        public static Field GetField(this UnaryExpression expression)
+        {
+            if (expression.Operand.IsMethodCall())
+            {
+                return expression.Operand.ToMethodCall().GetField();
+            }
+            throw new NotSupportedException($"Expression '{expression.ToString()}' is currently not supported.");
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Field"/> defined on the current instance of <see cref="MethodCallExpression"/>
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+        public static Field GetField(this MethodCallExpression expression)
+        {
+            if (expression.Object?.IsMember() == true)
+            {
+                return expression.Object.ToMember().GetField();
+            }
+            else
+            {
+                // Contains
+                if (expression.Method.Name == "Contains")
+                {
+                    var last = expression.Arguments?.Last();
+                    if (last?.IsMember() == true)
+                    {
+                        return last.ToMember().GetField();
+                    }
+                }
+            }
+            throw new NotSupportedException($"Expression '{expression.ToString()}' is currently not supported.");
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Field"/> defined on the current instance of <see cref="MemberExpression"/>
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+        public static Field GetField(this MemberExpression expression)
+        {
+            var member = expression.Member;
+            if (member.IsPropertyInfo())
+            {
+                return member.ToPropertyInfo().AsField();
+            }
+
+            if (member.IsFieldInfo())
+            {
+                var fieldInfo = member.ToFieldInfo();
+                return new Field(fieldInfo.Name,fieldInfo.FieldType);
+            }
+
```

Comment:
Perfect changes! :) Anyway, can we remove this empty lines at (143 and 149) and can we instead put the 'else if' instead of 'if'?

Then I will accept everything now.

I also showed you some coding standards and principles of my programming way.

Hoping we soon sync each other :) --- BTW, Big Thanks !!!

Reply:
looking forward to learning more! thank you!!

Reply:
You now understand the 'Expression' tree (a part of RepoDb parser in the compilation). :)